woocommerce/woocommerce
# Add link to error snackbar

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR adds the code to include a link to the error in the snackbar warning.

We are showing a snackbar when the form has an error, and the user presses `Publish`, `Save draft` or `Preview`. Now, we include a link to the error tab/section next to the error message.

![Screenshot 2024-06-10 at 10 49 19 AM](https://github.com/woocommerce/woocommerce/assets/1314156/63f0c47e-e4ec-41d2-8c12-610e7cc8759b)

This PR also includes the code to scroll directly to a section in the form when it is passed as a `block_id` in the URL.

Before showing the input with the error, we display the warning `Changes you made may not be saved.`, and I don't think we should show it. I think it would be good to tackle that in a follow-up PR. 

Closes #46768.

```[tasklist]
### ACs
- [ ] Add link to the snackbar pointing tab/section where the error originates
```



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure `Try the new product editor (Beta)` is enabled under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`.
2. Go to `Products > Add new` and add a Summary text.
3. Press `Publish`. You will see a snackbars with the message `Product name is required. View error`.
4. Since you are in the `General` tab and the product name field is visible, the snackbar will not have any links and will disappear automatically.
5. Go to the `Shipping` tab.
6. Repeat step 3. The same snackbar will be shown, but it will contain a link to the tab (and section) with the error. It won't dismiss automatically.
7. Try the link in the snackbar. It should redirect you to the tab/section with the error.
8. Go to the `Variations` tab and create some.
9. Edit a single variation.
10. Add a `Note` and go to the `Shipping` tab
11. Press `Update`.
12. You will see a snackbar with the message `Regular price is required.`, a link pointing to the error tab/section, and an `X` to dismiss it.
13. Verify that the link in the snackbar works as expected.
14. Go to the `General` tab and press `Update`.
15. The same snackbar will be shown without the link and the `X` to dismiss it.
16. Now try updating a variation with a duplicated SKU. You can do that under the `Inventory` tab.
17. Add an SKU for that variation; use `product`, for example.
18. Now edit another variation and use the same SKU.
19. Try to update the product, but stay on a tab other than `Inventory`. For example, stay on the `Shipping` tab and press `Update`.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
